pkg/intercp/tls/pki.go

```diff
@@ -0,0 +1,75 @@
+package tls
+
+import (
+	"context"
+	"crypto/rsa"
+	tls "crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/kumahq/kuma/pkg/core/resources/apis/system"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	util_tls "github.com/kumahq/kuma/pkg/tls"
+)
+
+var GlobalSecretKey = model.ResourceKey{
+	Name: "inter-cp-ca",
+}
+
+func GenerateCA() (*util_tls.KeyPair, error) {
+	subject := pkix.Name{
+		Organization:       []string{"Kuma"},
+		OrganizationalUnit: []string{"Mesh"},
+		CommonName:         "Control Plane Intercommunication CA",
+	}
+	return util_tls.GenerateCA(util_tls.DefaultKeyType, subject)
+}
+
+func LoadCA(ctx context.Context, resManager manager.ReadOnlyResourceManager) (tls.Certificate, error) {
+	globalSecret := system.NewGlobalSecretResource()
+	if err := resManager.Get(ctx, globalSecret, store.GetBy(GlobalSecretKey)); err != nil {
+		return tls.Certificate{}, err
+	}
+	bytes := globalSecret.Spec.GetData().GetValue()
+	certBlock, rest := pem.Decode(bytes)
+	keyBlock, _ := pem.Decode(rest)
```

Comment:
I don't have an expertise with this part of stdlib so this may be a silly question, but are we sure we can ignore an error here?

Reply:
I think we can. If someone wants to put random bytes at the end of two pem blocks, it's up to them I guess.
We only care if we can decode cert and key (and validate if this is actually cert and key)